packages/medusa/src/services/cart.ts

```diff
@@ -457,7 +463,22 @@ class CartService extends BaseService {
 
       await this.lineItemService_.withTransaction(manager).delete(lineItem.id)
 
-      const result = await this.retrieve(cartId)
+      const result = await this.retrieve(cartId, {
+        relations: ["items", "discounts", "discounts.rule"],
+      })
+
+      // delete all old line item adjustments
+      await this.lineItemAdjustmentService_.withTransaction(manager).delete({
+        item_id: result.items
+          .filter((i) => !i.is_return)
+          .map((item) => item.id),
+      })
+
+      // adjust existing line item adjustments
+      await this.lineItemAdjustmentService_
+        .withTransaction(manager)
+        .createAdjustments(result)
+
```

Comment:
nit: move this into a private CartService method `refreshAdjustments_(cartId)` as this is also used on line 597.